packages/backend/src/apps/formsg/triggers/new-submission/get-mock-data.ts

```diff
@@ -0,0 +1,100 @@
+import { IGlobalVariable } from '@plumber/types'
+
+import ObjectID from 'bson-objectid'
+import { DateTime } from 'luxon'
+
+import { getFormDetailsFromGlobalVariable } from '../../common/webhook-settings'
+
+type FormField = {
+  _id: string
+  title: string
+  fieldType: string
+  fieldOptions?: Array<string>
+  columns?: Array<{
+    _id: string
+    title: string
+    columnType: 'textfield' | 'dropdown'
+    fieldOptions?: Array<string>
+  }>
+}
+
+function getSampleValue(field: FormField) {
+  switch (field.fieldType) {
+    case 'textarea':
+    case 'textfield':
+      return 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.'
+    case 'radiobutton':
+    case 'dropdown':
+      return field.fieldOptions[0]
+    case 'checkbox':
+      return [field.fieldOptions[0]]
+    case 'country_region':
+      return 'SINGAPORE'
+    case 'date':
+      return DateTime.now().toFormat('dd MMM yyyy')
+    case 'table': {
+      const sampleRow: Array<string> = field.columns.map(
+        ({ columnType, fieldOptions, title, _id }) =>
+          getSampleValue({
+            _id,
+            title,
+            fieldType: columnType,
+            fieldOptions: fieldOptions,
+          }) as string,
+      )
+      return [sampleRow, sampleRow]
+    }
+    case 'email':
+      return 'hello@example.com'
+    case 'decimal':
+      return 1.234 as number
+    case 'number':
+      return 1234 as number
+    case 'mobile':
+      return '+6598765432'
+    case 'homeno':
+      return '+6567890123'
+    case 'yes_no':
+      return 'Yes'
+    case 'rating':
+      return '1'
+    default:
+      return ''
+  }
+}
+
+function getSampleAnswerKey(fieldType: string) {
+  if (fieldType === 'checkbox') {
+    return 'answerArray'
+  }
+  return 'answer'
+}
+
+async function getMockData($: IGlobalVariable) {
+  try {
+    const { formId } = getFormDetailsFromGlobalVariable($)
+    const { data } = await $.http.get(`/v3/forms/${formId}`)
```

Comment:
We can use `https://form.gov.sg/api/v3/forms/${formId}/sample-submission` API to get sample submissions, this is the same as what they used in the zapier integration

Reply:
This endpoint is not ready yet me thinks. Was crashing their server jn when I tried it